Test mpOpenAPI config with checkpoint/restore

anjumfatima90 opened the pull request:
for #26966

anjumfatima90 committed:
Test mpOpenAPI config with checkpoint/restore

tjwatson commented on dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/OpenAPIConfigTest.java:
```diff
@@ -0,0 +1,174 @@
+    @BeforeClass
+    public static void deployApp() throws Exception {
+        // Set guards
+        server.setJvmOptions(Arrays.asList("-Dcom.ibm.ws.beta.edition=true"));
```
We will need to remember remove this JVM option.

tjwatson commented on dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/OpenAPIConfigTest.java:
```diff
@@ -0,0 +1,174 @@
+    public static LibertyServer server;
+
+    @ClassRule
+    public static RepeatTests r = FATSuite.defaultMPRepeat(SERVER_NAME);
```
Can this really repeat.  I thought the settings tested here were specific to mpOpenApi-3.1

anjumfatima90 commented:
#build

anjumfatima90 committed:
Set post stop server archive to false while stopping the server before updating the server config

anjumfatima90 commented:
#build

anjumfatima90 committed:
Ignore ISE while stopping the server

The tests do a stop server before updating the config. This stops the WAB io.openliberty.microprofile.openapi.2.0.internal.servlet.
But sometimes it throws ISE: BundleContext is no longer valid,  intermittently when ungetting the service because the service is chenged and the service tracker tries to unget the service and notices that the bundle context was closed and invalid.

Updated Equinox would no longer throw exceptions on invalid context when ungetService is called.
For reference: https://github.com/eclipse-equinox/equinox/issues/172

tjwatson commented on dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/OpenAPIConfigTest.java:
```diff
@@ -89,7 +89,7 @@ public void setup() throws Exception {
     public void teardown() throws Exception {
         try {
-            server.stopServer();
+            server.stopServer("CWWKE0701E");
```
This is pretty general.  I was thinking you would use regex on the specific message:

```java
"CWWKE0701E: .*BundleContext is no longer valid io.openliberty.microprofile.openapi.2.0.internal.servlet"
```

anjumfatima90 commented:
#libby
